### PR TITLE
Update python support to 3.10, 3.11 and 3.12

### PR DESCRIPTION
**Pull request by sjdilkes**
This PR drops python 3.9 support and adds python 3.12.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: .github/workflows/build_and_test.yml
Comment (cqc-alec) [inline .github/workflows/build_and_test.yml:1]: Prefer to map (3,9, 3.10, 3.11) --> (3.10, 3.11, 3.12) in this workflow. (So 3.12 will be tested in this PR.)

**Review by cqc-alec — CHANGES_REQUESTED** [review]
And changelog,..

File: .github/workflows/build_and_test.yml
Comment (sjdilkes) [reply .github/workflows/build_and_test.yml:1]: I've updated this file to do this mapping

**Comment by cqc-alec** [comment]
Looks like some dependency requires numpy 1.23.3, which is failing to build on Windows. May have to skip 3.12 tests on Windows for the time being?

File: README.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ Some useful links:
 
 ## Getting started
 
-`pytket-braket` is available for Python 3.9, 3.10 and 3.11, on Linux, MacOS
+`pytket-braket` is available for Python 3.10, 3.11 and 3.12, on Linux, MacOS
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply README.md]: added

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -37,48 +37,43 @@ jobs:
...
       with:
-        python-version: '3.11'
-    - name: Build and test (3.11) - macos and ubuntu
+        python-version: '3.12'
+    - name: Build and test (3.12) - macos and ubuntu
       if: (matrix.os != 'windows-2022') && (github.event_name == 'push' || github.event_name == 'pull_request' || github.event_name == 'schedule')
       shell: bash
       run: |
         ./.github/workflows/build-test nomypy numpy
-    - name: Build and test (3.11) - windows
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply .github/workflows/build_and_test.yml:77]: We still should be:
```- name: Build and test including remote checks (3.11) nomypy
      if:  (matrix.os != 'macos-12') && (github.event_name == 'push' || (github.event_name == 'pull_request' &&  github.event.pull_request.head.repo.full_name == github.repository) || github.event_name == 'release' || github.event_name == 'schedule')```
      
I think this is just an artifact of doing the 3.9->3.10, 3.10->3.11, 3.11->3.12 mapping.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**